interp/constrexpr_ops.ml

```diff
@@ -352,14 +352,13 @@ let coerce_reference_to_id = function
         (str "This expression should be a simple identifier.")
 
 let coerce_to_id = function
-  | _loc, CRef (Ident (loc,id),_) -> (loc,id)
-  | a -> CErrors.user_err ?loc:(constr_loc a)
+  | { CAst.v = CRef (Ident (loc,id),_); _ } -> (loc,id)
+  | { CAst.loc; _ } -> CErrors.user_err ?loc
        ~hdr:"coerce_to_id"
          (str "This expression should be a simple identifier.")
 
 let coerce_to_name = function
-  | _loc, CRef (Ident (loc,id),_) -> (loc,Name id)
-  | loc, CHole (_,_,_) -> (loc,Anonymous)
-  | a -> CErrors.user_err
-         ?loc:(constr_loc a) ~hdr:"coerce_to_name"
-         (str "This expression should be a name.")
+  | { CAst.v = CRef (Ident (loc,id),_) } -> (loc,Name id)
+  | { CAst.loc; CAst.v = CHole (_,_,_) } -> (loc,Anonymous)
```

Comment:
note: if you qualify the first field of the record, you don't need to qualify the rest, so `{ CAst.loc; v = ... }` works. (Since 4.04.0, the more regular `CAst.{ loc; v = ... }` is also accepted.)